azure-proto-core/ArmClient.cs

```diff
@@ -186,17 +192,20 @@ public ResourceGroupOperations ResourceGroup(PhResourceGroup resourceGroup)
             return null;
         }
 
-        public ArmResponse<ResourceOperationsBase<T>> CreateResource<T>(string subscription, string resourceGroup, string name, T model, azure_proto_core.Location location = default) where T:TrackedResource
+        public ArmResponse<TOperations> CreateResource<TContainer, TOperations, TResource>(string subscription, string resourceGroup, string name, TResource model, azure_proto_core.Location location = default)
+            where TResource : TrackedResource
+            where TOperations : ResourceOperationsBase<TOperations, TResource>
+            where TContainer : ResourceContainerOperations<TOperations, TResource>
         {
             if (location == null)
             {
                 location = azure_proto_core.Location.Default;
             }
 
-            ResourceContainerOperations<ResourceOperationsBase<T>, T> container;
-            if (!Registry.TryGetContainer(this.ClientContext, new ArmResource($"/subscriptions/{subscription}/resourceGroups/{resourceGroup}", location), out container))
+            TContainer container;
+            if (!Registry.TryGetContainer<TContainer, ArmResource, TOperations, TResource>(this.ClientContext, new ArmResource($"/subscriptions/{subscription}/resourceGroups/{resourceGroup}", location), out container))
```

Comment:
At this point, do we really need the registry to do this?  Can't we use Activator.CreateInstance to instantiate the Container class and then call the create method with the given arguments?

Feels like we would only need the registry if we wanted to provide only the resource type and arguments.